posts/2022/05/the-evolution-of-the-scipy-developer-cli.md

```diff
@@ -0,0 +1,170 @@
+<!--
+.. title: The evolution of the SciPy developer CLI
+.. slug: the-evolution-of-the-scipy-developer-cli
+.. date: 2022-05-03 8:00:00 UTC+05:30
+.. author: Sayantika Banik
+.. tags: CLI, SciPy, doit, Typer, Rich, Click
+.. category:
+.. link:
+.. description:
+.. type: text
+.. previewimage:
+-->
+
+#### 🤔 What is a command-line interface (CLI)?
+
+Imagine a situation where there is a massive system with various tools and functionality. Every functionality requires a special command or an input from the user, CLI is designed to tackle such situations. Like a catalog it lists all the options present, helping the user to navigate.
+
+<p align="center">
+  <img alt="CLI example" src="https://user-images.githubusercontent.com/17350312/166633508-a2795c44-30bc-4a5b-8043-65beab71d31f.png" />
+</p>
+
+Now that we understand what a `CLI` is, how about we dive into the world of `SciPy`?
+
+<!-- TEASER_END -->
+
+#### 🤓 SciPy CLI journey
+
+An open-source project undergoes multiple iterations, and each iteration adds layers of tools and functionalities to incorporate. Right from the initial build, test, benchmarking, release notes, and beyond. Contribution guidelines start spanning across multiple pages, exponentially increasing efforts to maintain.
```

Comment:
I think the word incorporate in  the first sentence doesn't quite make sense to me. Maybe `to it` simply?

The next line is also  a bit tricky. I think it's missing a verb? or maybe it's supposed to connec to the next one somehow? but on it's own I can't quite make sense of it...

For the last line I'd make suggest a tweak: `Over time, the contribution guides start to span over more and more pages, as the over all effort to maintain the project grows exponentially" 
(I don't actually know if the growth is exponential, I'm taking your lead here 😉 )

Reply:
Over time, the contribution guides start to span over more and more pages, `as` the over all effort to maintain the project grows exponentially

Thanks for the suggestion :)
Perhaps we could replace `as` with `and`, the result and outcome sound opposite here.

Reply:
both work!
`as` makes a causal links. the guide is growing because the the maintenance effort is grow. 
`and` just means both these things happen. no causality implied.